cpp/src/column/column.cu

```diff
@@ -244,7 +247,26 @@ struct create_column_from_view {
             std::enable_if_t<std::is_same<ColumnType, cudf::struct_view>::value> * = nullptr>
   std::unique_ptr<column> operator()()
   {
-    CUDF_FAIL("struct_view not supported yet");
+    if (view.is_empty()) { return cudf::empty_like(view); }
+
+    std::vector<std::unique_ptr<column>> children;
+    children.reserve(view.num_children());
+
+    std::transform(view.child_begin(),
+                   view.child_end(),
+                   std::back_inserter(children),
+                   [stream = this->stream, mr = this->mr](auto child) {
+                     return std::make_unique<column>(child, stream, mr);
+                   });
+
+    auto num_rows = children.empty() ? 0 : children.front()->size();
+
+    return make_structs_column(num_rows,
+                               std::move(children),
+                               view.null_count(),
+                               cudf::copy_bitmask(view.null_mask(), 0, view.size(), stream, mr),
+                               stream,
+                               mr);
```

Comment:
Does making a structs column involve any device memory allocation? If not, is the `mr` parameter passed just for consistency?

Reply:
Gotcha. Thanks!